connector.rb

```diff
@@ -1,37 +1,2789 @@
+          {
+            name: 'webhook_identifier',
+            label: 'Webhook identifier',
+            optional: false,
+            hint: 'The webhook GraphQL identifier. An array of GraphQL identifiers can be set via "Formula" mode.'
+          },
+          {
+            name: 'webhook_policy',
+            label: 'JWT algorithm',
+            control_type: 'select',
+            pick_list: 'webhook_jwt',
+            optional: false,
+            hint: 'The webhook policy JWT algorithm.'
+          },
+          {
+            ngIf: 'input.webhook_policy == "HS256" || input.webhook_policy == "HS384" || ' \
+                  'input.webhook_policy == "HS512"',
+            name: 'webhook_secret_hmac',
+            label: 'HMAC secret',
+            control_type: 'password',
+            hint: 'The webhook policy HMAC secret.'
+          },
+          {
+            ngIf: 'input.webhook_policy == "RS256" || input.webhook_policy == "RS384" || ' \
+                  'input.webhook_policy == "RS512" || input.webhook_policy == "ES256" || ' \
+                  'input.webhook_policy == "ES256" || input.webhook_policy == "ES384" || ' \
+                  'input.webhook_policy == "ES512"',
+            name: 'webhook_public_key',
+            label: 'Public PEM key',
+            control_type: 'text-area',
+            hint: 'The webhook policy public key.'
+          },
+          {
+            ngIf: 'input.event_selection == "automation_rule"',
+            name: 'payload_output',
+            extends_schema: true,
+            schema_neutral: true,
+            control_type: 'schema-designer',
+            label: 'Payload schema',
+            sticky: true,
+            hint: 'The payload schema as defined in the "automation_rule" webhook event.'
+          }
+        ],
+
+      webhook_key: lambda do |connection, input|
+        'default_route'
+      end,
 
+      webhook_notification: lambda do |
+        input,
+        payload,
+        extended_input_schema,
+        extended_output_schema,
+        headers,
+        params,
+        connection,
+        webhook_subscribe_output|
+
+        jwt = payload&.[]('jwt')
+        if jwt.present?
+          data = case input['webhook_policy']
+                 when /^HS/
+                   workato.jwt_decode(jwt, input['webhook_secret_hmac'], input['webhook_policy'])['payload']['data']
+                 when /^RS/
+                   workato.jwt_decode(jwt, input['webhook_public_key'], input['webhook_policy'])['payload']['data']
+                 when /^ES/
+                   workato.jwt_decode(jwt, input['webhook_public_key'], input['webhook_policy'])['payload']['data']
+                 end
+
+          webhook_identifiers = input['webhook_identifier']
+          webhook_identifiers = Array.wrap(webhook_identifiers)
+
+          if webhook_identifiers.include?(data['webhook_nodeID']) &&
+             (data['event'] == input['event_selection'] || data['event'] == 'webhook.verify')
```

Comment:
Question raised during internal review: 
Is it possible to move this check to `webhook_key`/`webhook_keys` so it will be more efficient routing to specific webhooks, not all of them?

`webhook_keys` block does not support `workato.jwt_decode` but it supports regular methods on strings `decode_base64` and `split`. 

Do you think it is possible to parse payload for routing without signature verification. In webhook_notification you still need to make proper decode.

Reply:
@den-sergeyev-workato 
Yes, I can make the changes.
One question, when I make the change then I also lose the functionality of specifying multiple webhook identifiers?
The result of the `webhook_key` should be an exact match with the value returned in `webhook_keys`. In the `webhook_notification` now use `webhook_identifiers.include?(data['webhook_nodeID'])`. The webhook input in formula mode allows me to input an array instead of a string.

Reply:
@den-sergeyev-workato
I spend quite some time changing the webhook routing and but got it working, will be in the next commit.